guides/source/association_basics.md

```diff
@@ -7,20 +7,56 @@ This guide covers the association features of Active Record.
 
 After reading this guide, you will know how to:
 
+* Understand the various types of associations.
 * Declare associations between Active Record models.
-* Understand the various types of Active Record associations.
-* Use the methods added to your models by creating associations.
+* Choose the right association type for your models.
+* Use Single Table Inheritance.
+* Setting up and using Delegated Types.
 
 --------------------------------------------------------------------------------
 
-Why Associations?
+Associations Overview
 -----------------
 
-In Rails, an _association_ is a connection between two Active Record models. Why do we need associations between models? Because they make common operations simpler and easier in your code.
+_Associations_ in Rails are special macro style calls that make it easy to
+connect different models. You can use them to tell Rails how your models relate
+to each other, which helps you manage your data more effectively, and makes
+common operations simpler and easier to read.
 
-For example, consider a simple Rails application that includes a model for authors and a model for books. Each author can have many books.
+INFO: A macro-style call is a method that generates or modifies other methods at
+runtime, allowing for concise and expressive declarations of functionality, such
+as defining model associations in Rails. For example, `has_many :comments`.
 
-Without associations, the model declarations would look like this:
+When you set up an association, you instruct Rails to maintain the [Primary
+Key](https://en.wikipedia.org/wiki/Primary_key) and [Foreign
+Key](https://en.wikipedia.org/wiki/Foreign_key) information between instances of
+the two models, ensuring that your data stays consistent and properly linked.
+This makes it easy to keep track of which records are related. It also adds
+useful methods to your models so you can work with related data more easily.
+
+Consider a simple Rails application with models for authors and books.
+
+### Without Associations
+
+Without associations, creating and deleting books for that author would require
+a tedious and manual process. Here's what that would look like:
+
+```ruby
+class CreateAuthors < ActiveRecord::Migration[7.2]
+  def change
+    create_table :authors do |t|
+      t.string :name
+      t.timestamps
+    end
+
+    create_table :books do |t|
+      t.belongs_to :author
```

Comment:
You may want to use its alias `t.references` to avoid possible confusion with `belongs_to` in ActiveRecord classes.

Reply:
Should we mention somewhere that these are aliases?

Reply:
Also, whatever we pick, let's keep it consistent throughout the page.

Reply:
> Should we mention somewhere that these are aliases?

Since this is related to Active Record Migrations, folks will refer to that guide to learn more about migrations and [we mention that `belongs_to` is an alias there. ](https://guides.rubyonrails.org/active_record_migrations.html#creating-associations)

Reply:
> You may want to use its alias t.references to avoid possible confusion with belongs_to in ActiveRecord classes.

I think that the `belongs_to` alias is explicit and makes the migrations easier to read, however, I see your point about possible confusion here and so I've changed it to references but I hope that it's not misleading. 

> Also, whatever we pick, let's keep it consistent throughout the page.

@MatheusRich - my ears perk up when folks say we should keep it consistent because I agree wholeheartedly! In this case, I've used `t.belongs_to` in all the scenarios except the one that @rizwanreza pointed out because even though we can use the alias in this case, I agree that it may cause confusion.